integration/insert_compat_test.go

```diff
@@ -112,16 +117,54 @@ func TestInsertCompat(t *testing.T) {
 
 	testCases := map[string]insertCompatTestCase{
 		"InsertEmptyDocument": {
-			insert: bson.D{},
+			insert: []any{bson.D{}},
+			// TODO: this test fails with last byte being different as the value is always unique,
+			// see https://github.com/mongodb/mongo-go-driver/blob/e1bf8858dd9ba111e880f8e5a8b7e7b7da64cb54/bson/primitive/objectid.go#L51
+			// Target: &mongo.InsertManyResult{InsertedIDs:[]interface {}{primitive.ObjectID{0x63, 0xac, 0x15, 0x9e, 0xe6, 0x49, 0x0, 0x8, 0x7, 0x15, 0x28, 0x9a}}}
+			// Compat: &mongo.InsertManyResult{InsertedIDs:[]interface {}{primitive.ObjectID{0x63, 0xac, 0x15, 0x9e, 0xe6, 0x49, 0x0, 0x8, 0x7, 0x15, 0x28, 0x9b}}}
```

Comment:
@AlekSi this is the test case I mentioned (the one where `_id` for target and compat are different because the driver inserts a unique id every time). I suggest we just shouldn't compare if `_id`s are the same in this case.